net/netcheck/netcheck.go

```diff
@@ -125,6 +128,44 @@ type Report struct {
 	// TODO: update Clone when adding new fields
 }
 
+// GetGlobalAddrs returns the v4 and v6 global addresses observed during the
+// netcheck, which includes the best latency endpoint first, followed by any
+// other endpoints that were observed repeatedly. It excludes singular endpoints
+// that are likely only the result of a hard NAT.
+func (r *Report) GetGlobalAddrs() ([]netip.AddrPort, []netip.AddrPort) {
+	var v4, v6 []netip.AddrPort
+	// Always add the best latency entries first.
+	if r.GlobalV4 != "" {
+		v4 = append(v4, netip.MustParseAddrPort(r.GlobalV4))
```

Comment:
... which would save the parsing in this function (and the other 3 cases below).